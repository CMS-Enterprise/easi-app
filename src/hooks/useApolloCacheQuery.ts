// useApolloCacheQuery.ts
import {
  DocumentNode,
  OperationVariables,
  QueryHookOptions,
  QueryResult,
  TypedDocumentNode,
  useApolloClient,
  useQuery
} from '@apollo/client';

/** Custom hook to query the cache to determine if query result exists already */
export default function useApolloCacheQuery<
  TData = any,
  TVariables = OperationVariables
>(
  query: DocumentNode | TypedDocumentNode<TData, TVariables>,
  options?: QueryHookOptions<TData, TVariables>
): QueryResult<TData, TVariables> {
  const { cache } = useApolloClient();

  const queryResult = useQuery<TData, TVariables>(query, options);

  return {
    ...queryResult,
    loading:
      !cache.diff({
        query: cache.transformDocument(query),
        variables: options?.variables,
        returnPartialData: true,
        optimistic: false
      }).complete && queryResult.loading
  };
}
