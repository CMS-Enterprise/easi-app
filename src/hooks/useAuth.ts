import { useSelector } from 'react-redux';
import { useLazyQuery } from '@apollo/client';
import { useFlags } from 'launchdarkly-react-client-sdk';

import {
  BASIC_USER_PROD,
  GOVTEAM_DEV,
  GOVTEAM_PROD,
  TRB_ADMIN_DEV,
  TRB_ADMIN_PROD
} from 'constants/jobCodes';
import GetSystemIntakeGrbReviewersQuery from 'queries/GetSystemIntakeGrbReviewersQuery';
import {
  GetSystemIntakeGrbReviewers,
  GetSystemIntakeGrbReviewersVariables
} from 'queries/types/GetSystemIntakeGrbReviewers';
import { AppState } from 'reducers/rootReducer';

type UseAuthReturn = {
  /**
   * Object with helper functions to check job codes / roles
   *
   * Returns undefined if user is not set
   */
  user:
    | {
        isTrbAdmin: () => boolean;
        isGrtReviewer: () => boolean;
        isBasicUser: () => boolean;
        isGrbReviewer: (systemIntakeId: string) => boolean;
      }
    | undefined;
  /** Returns true if function with query is called and loading */
  loading: boolean;
};

/** Hook that returns functions to check for roles / job codes */
const useAuth = (): UseAuthReturn => {
  const auth = useSelector((state: AppState) => state.auth);
  const { groups, euaId } = auth;

  const flags = useFlags();

  const [getGrbReviewers, { data, called, loading }] = useLazyQuery<
    GetSystemIntakeGrbReviewers,
    GetSystemIntakeGrbReviewersVariables
  >(GetSystemIntakeGrbReviewersQuery);

  const isUserSet = auth.isUserSet && !loading;

  /** Check if user is TRB admin */
  const isTrbAdmin = (): boolean => {
    if (flags.downgradeTrbAdmin) {
      return false;
    }

    if (groups.includes(TRB_ADMIN_DEV) || groups.includes(TRB_ADMIN_PROD)) {
      return true;
    }

    return false;
  };

  /** Check if user is GrtReviewer */
  const isGrtReviewer = (): boolean => {
    if (flags.downgradeGovTeam) {
      return false;
    }

    if (groups.includes(GOVTEAM_DEV) || groups.includes(GOVTEAM_PROD)) {
      return true;
    }

    return false;
  };

  /** Check if basic user */
  const isBasicUser = (): boolean => {
    if (groups.includes(BASIC_USER_PROD)) {
      return true;
    }
    if (groups.length === 0) {
      return true;
    }
    if (!isGrtReviewer() && !isTrbAdmin()) {
      return true;
    }
    return false;
  };

  /** Check if user is set as system intake GRB reviewer */
  const isGrbReviewer = (systemIntakeId: string) => {
    if (!called) {
      getGrbReviewers({
        variables: { id: systemIntakeId }
      });
    }

    const grbReviewers = data?.systemIntake?.grbReviewers;

    if (!grbReviewers || grbReviewers.length === 0) {
      return false;
    }

    return grbReviewers.some(
      ({ userAccount }) => userAccount.username === euaId
    );
  };

  const user: UseAuthReturn['user'] = {
    isTrbAdmin,
    isGrtReviewer,
    isBasicUser,
    isGrbReviewer
  };

  return {
    user: isUserSet || !loading ? user : undefined,
    loading
  };
};

export default useAuth;
