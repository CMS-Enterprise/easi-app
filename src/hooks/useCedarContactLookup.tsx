/** Custom hook for retrieving contacts from Cedar by common name */

import { useMemo, useState } from 'react';
import { ApolloQueryResult, useQuery } from '@apollo/client';

import GetCedarContactsQuery from 'queries/GetCedarContactsQuery';
import { GetCedarContacts } from 'queries/types/GetCedarContacts';
import { CedarContactProps } from 'types/systemIntake';

function useCedarContactLookup(): [
  CedarContactProps[],
  (commonName: string) => void
];

function useCedarContactLookup(
  query: string,
  euaUserId: string
): CedarContactProps | undefined;

function useCedarContactLookup(
  query?: string,
  euaUserId?: string
):
  | [CedarContactProps[], (commonName: string) => void]
  | CedarContactProps
  | undefined {
  const { data, refetch } = useQuery<GetCedarContacts>(GetCedarContactsQuery, {
    variables: { commonName: query },
    skip: !query
  });
  const [contacts, setContacts] = useState<CedarContactProps[]>(
    data?.cedarPersonsByCommonName || []
  );

  const contactByEuaUserId = useMemo<CedarContactProps | undefined>(() => {
    return data?.cedarPersonsByCommonName.find(
      contact => contact.euaUserId === euaUserId
    );
  }, [data?.cedarPersonsByCommonName, euaUserId]);

  const queryCedarContacts = (commonName: string) => {
    // Removes eua id from "John Doe, ABCD" format when querying
    const commonNameQuery = commonName.trim().split(',')[0];

    if (commonNameQuery.length > 1) {
      refetch({ commonName: commonNameQuery })
        .then((response: ApolloQueryResult<GetCedarContacts>) => {
          setContacts(
            sortCedarContacts(
              response?.data?.cedarPersonsByCommonName,
              commonNameQuery
            )
          );
        })
        .catch(() => {
          // If error, set contacts to empty array
          setContacts([]);
        });
    }
  };

  return euaUserId ? contactByEuaUserId : [contacts, queryCedarContacts];
}

const sortCedarContacts = (
  contacts: CedarContactProps[],
  query: string
): CedarContactProps[] => {
  return [...contacts].sort((a, b) => {
    const result =
      a.commonName.toLowerCase().search(query) -
      b.commonName.toLowerCase().search(query);
    if (result > 0) return 1;
    if (result < 0) return -1;
    return 0;
  });
};

export default useCedarContactLookup;
