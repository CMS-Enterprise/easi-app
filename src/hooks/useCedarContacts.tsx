import { useCallback, useMemo } from 'react';
import { useQuery } from '@apollo/client';

import GetCedarContactsQuery from 'queries/GetCedarContactsQuery';
import { GetCedarContacts } from 'queries/types/GetCedarContacts';
import { CedarContactProps } from 'types/systemIntake';

export default function useCedarContacts(): {
  contacts: CedarContactProps[] | null;
  getContactByEua: (euaUserId: string) => CedarContactProps | null;
} {
  const { data } = useQuery<GetCedarContacts>(GetCedarContactsQuery, {
    variables: { commonName: '' }
  });
  const contacts = useMemo(() => data?.cedarPersonsByCommonName || null, [
    data?.cedarPersonsByCommonName
  ]);
  const getContactByEua = useCallback(
    (euaUserId: string) => {
      return (
        (contacts || []).find(contact => contact.euaUserId === euaUserId) ??
        null
      );
    },
    [contacts]
  );
  return { contacts, getContactByEua };
}
