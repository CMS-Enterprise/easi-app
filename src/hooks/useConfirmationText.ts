import { useEffect, useState } from 'react';
import { useHistory, useLocation } from 'react-router-dom';

// useConfirmationText is a hook used to get the confirmation text
// passed through react router state.
// const confirmationText = useConfirmationText();
//
// Two common ways to pass state through react-router (there are more)
// history.push('/some/url', { confirmationText: 'Confirmed' });
// or
// const location = { pathname: '/some/url', state: { confirmationText: 'Confirmed' }};
// <Link to={location} />
function useConfirmationText() {
  const location = useLocation<any>();
  const history = useHistory();
  const [confirmationText, setConfirmationText] = useState('');

  useEffect(() => {
    if (location.state && location.state.confirmationText) {
      setConfirmationText(location.state.confirmationText);
      history.replace({
        state: {}
      });
    }

    // unset confirmationText if it's null
    if (!location.state?.confirmationText) {
      setConfirmationText('');
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [history.location.pathname]);

  return confirmationText;
}

export default useConfirmationText;
