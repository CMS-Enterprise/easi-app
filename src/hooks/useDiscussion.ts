import { useHistory, useLocation } from 'react-router-dom';

export type DiscussionMode = 'view' | 'start' | 'reply' | undefined;

export default function useDiscussion() {
  const history = useHistory();
  const location = useLocation();

  return {
    getDiscussionMode() {
      const q = new URLSearchParams(location.search);
      const discussionMode = (q.get('discussion') ||
        undefined) as DiscussionMode;
      return discussionMode;
    },
    getDiscussionId() {
      const q = new URLSearchParams(location.search);
      const discussionId = q.get('discussionId') || undefined;
      return discussionId;
    },
    setDiscussion(query: string | false) {
      if (query === false) {
        history.push(`${location.pathname}`);
        return;
      }
      history.push(`${location.pathname}?${query}`);
    }
  };
}
