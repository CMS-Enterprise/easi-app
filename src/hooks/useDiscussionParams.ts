import { useHistory, useLocation } from 'react-router-dom';
import { SystemIntakeGRBDiscussionBoardType } from 'gql/generated/graphql';

const discussionModeKeys = ['view', 'start', 'reply'] as const;

export type DiscussionMode = (typeof discussionModeKeys)[number];

export type UseDiscussionParamsReturn = {
  /**
   * Get discussion board query parameters
   *
   * Undefined `discussionMode` implies closed modal
   */
  getDiscussionParams: () => {
    discussionBoardType: SystemIntakeGRBDiscussionBoardType | undefined;
    discussionMode: DiscussionMode | undefined;
    discussionId: string | undefined;
  };
  /**
   * Push a new url query to update the Discussion subviews state.
   *
   * `false` implies closing the modal
   */
  pushDiscussionQuery: (
    query:
      | {
          discussionMode: Extract<DiscussionMode, 'view' | 'start'>;
          discussionBoardType: SystemIntakeGRBDiscussionBoardType;
        }
      | {
          discussionMode: Extract<DiscussionMode, 'reply'>;
          discussionBoardType: SystemIntakeGRBDiscussionBoardType;
          discussionId: string;
        }
      | false
  ) => void;
};

function discussionModeIsValid(
  discussionMode: DiscussionMode | string | null
): discussionMode is DiscussionMode {
  return (
    discussionMode !== null &&
    discussionModeKeys.includes(discussionMode as DiscussionMode)
  );
}

function discussionBoardTypeIsValid(
  discussionBoardType: SystemIntakeGRBDiscussionBoardType | string | null
): discussionBoardType is SystemIntakeGRBDiscussionBoardType {
  return (
    discussionBoardType !== null &&
    discussionBoardType in SystemIntakeGRBDiscussionBoardType
  );
}

/**
 * Handle Discussion (side panel modal) state with the url query params
 * `discussionMode` and `discussionId`.
 */
export default function useDiscussionParams(): UseDiscussionParamsReturn {
  const history = useHistory();
  const location = useLocation();

  const getDiscussionParams: UseDiscussionParamsReturn['getDiscussionParams'] =
    () => {
      const q = new URLSearchParams(location.search);

      const discussionMode = q.get('discussionMode');
      const discussionId = q.get('discussionId');

      const discussionBoardType =
        q.get('discussionBoardType') ||
        SystemIntakeGRBDiscussionBoardType.PRIMARY;

      // Silent fail on invalid query params
      if (
        !discussionModeIsValid(discussionMode) ||
        !discussionBoardTypeIsValid(discussionBoardType)
      ) {
        return {
          discussionBoardType: undefined,
          discussionMode: undefined,
          discussionId: undefined
        };
      }

      // Check reply mode for valid `discussionId`
      // Silent fail if `discussionId` is invalid
      if (discussionMode === 'reply') {
        if (discussionId === null)
          return {
            discussionBoardType: undefined,
            discussionMode: undefined,
            discussionId: undefined
          };

        return {
          discussionMode,
          discussionBoardType,
          discussionId
        };
      }

      return { discussionMode, discussionBoardType, discussionId: undefined };
    };

  const pushDiscussionQuery: UseDiscussionParamsReturn['pushDiscussionQuery'] =
    query => {
      if (query === false) {
        history.push(`${location.pathname}`);
        return;
      }

      const querystring = new URLSearchParams(query);
      history.push(`${location.pathname}?${querystring}`);
    };

  return {
    getDiscussionParams,
    pushDiscussionQuery
  };
}
