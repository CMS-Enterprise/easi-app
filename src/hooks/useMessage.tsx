import React, {
  createContext,
  ReactNode,
  useContext,
  useEffect,
  useState
} from 'react';
import { useLocation } from 'react-router-dom';

import Alert, { AlertProps } from 'components/shared/Alert';

type MessageProps = {
  message?: string | React.ReactElement;
  alertProps?: Omit<AlertProps, 'children'>;
};

type SetMessage = (
  message?: string | React.ReactElement,
  alertProps?: Omit<AlertProps, 'children'>
) => void;

const MessageContext = createContext<
  | {
      Message: ({ className }: { className?: string }) => JSX.Element | null;
      showMessage: SetMessage;
      showMessageOnNextPage: SetMessage;
    }
  | undefined
>(undefined);

// MessageProvider manages the state necessary for child components to
// use the useMessage hook.
const MessageProvider = ({ children }: { children: ReactNode }) => {
  const [queuedMessage, setQueuedMessage] = useState<MessageProps>({});
  const [message, setMessage] = useState<MessageProps>({});

  const location = useLocation();

  const [lastPathname, setLastPathname] = useState(location.pathname);

  const Message = (props: { className?: string }) => {
    // If no message, return null
    if (!message.message) return null;

    // If message has alert props, wrap in component
    if (message.alertProps) {
      return (
        <Alert {...message.alertProps} {...props}>
          {message.message}
        </Alert>
      );
    }

    // Return plain message
    return <>{message.message}</>;
  };

  useEffect(() => {
    if (lastPathname !== location.pathname) {
      setMessage(queuedMessage);
      setQueuedMessage({});
      setLastPathname(location.pathname);
    }
  }, [message, queuedMessage, lastPathname, location.pathname]);

  return (
    <MessageContext.Provider
      value={{
        Message,
        showMessage: (value, alertProps) =>
          setMessage({
            message: value,
            alertProps
          }),
        showMessageOnNextPage: (value, alertProps) =>
          setQueuedMessage({
            message: value,
            alertProps
          })
      }}
    >
      {children}
    </MessageContext.Provider>
  );
};

// useMessage is a hook that provides the ability to show notifications to the
// user. Components will use the hook to get access to two methods:
//
// showMessage: display a message to the user immediately. This is useful
// when an interaction won't cause an immediate redirect to a new page.
//
// showMessageOnNextPage: schedule a message to be shown on the next
// visited by the user. Pages are identified using their URL and therefore
// messages won't appear until the URL changes.
//
// Currently there is only support for showing a single message.
const useMessage = () => {
  const context = useContext(MessageContext);
  if (context === undefined) {
    throw new Error('useMessage must be used within a MessageProvider');
  }

  return context;
};

export default useMessage;
export { useMessage, MessageProvider };
