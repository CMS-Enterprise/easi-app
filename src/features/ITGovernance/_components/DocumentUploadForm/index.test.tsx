import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render, screen } from '@testing-library/react';

import { MessageProvider } from 'hooks/useMessage';
import easiMockStore from 'utils/testing/easiMockStore';

import DocumentUploadForm from '.';

describe('DocumentUploadForm page', () => {
  it('renders send notification check if viewing in admin context', async () => {
    const mockStore = easiMockStore({
      groups: [
        'EASI_TRB_ADMIN_D',
        'EASI_TRB_ADMIN_P',
        'EASI_D_GOVTEAM',
        'EASI_P_GOVTEAM'
      ]
    });

    render(
      <MemoryRouter>
        <MockedProvider>
          <Provider store={mockStore}>
            <MessageProvider>
              <DocumentUploadForm type="admin" />
            </MessageProvider>
          </Provider>
        </MockedProvider>
      </MemoryRouter>
    );

    expect(
      await screen.findByText(
        'Would you like to send a notification email to GRB reviewers?'
      )
    );
  });

  it('does not render send notification check if not viewing in admin context (even if admin perms)', async () => {
    const mockStore = easiMockStore({
      groups: [
        'EASI_TRB_ADMIN_D',
        'EASI_TRB_ADMIN_P',
        'EASI_D_GOVTEAM',
        'EASI_P_GOVTEAM'
      ]
    });

    render(
      <MemoryRouter>
        <MockedProvider>
          <Provider store={mockStore}>
            <MessageProvider>
              <DocumentUploadForm type="requester" />
            </MessageProvider>
          </Provider>
        </MockedProvider>
      </MemoryRouter>
    );

    expect(
      screen.queryByText(
        'Would you like to send a notification email to GRB reviewers?'
      )
    ).not.toBeInTheDocument();
  });

  it('does not render send notification check if viewing as a non-admin in requester context', async () => {
    const mockStore = easiMockStore({
      groups: ['EASI_P_USER']
    });

    render(
      <MemoryRouter>
        <MockedProvider>
          <Provider store={mockStore}>
            <MessageProvider>
              <DocumentUploadForm type="requester" />
            </MessageProvider>
          </Provider>
        </MockedProvider>
      </MemoryRouter>
    );

    expect(
      screen.queryByText(
        'Would you like to send a notification email to GRB reviewers?'
      )
    ).not.toBeInTheDocument();
  });

  it('does not render send notification check if viewing as a non-admin in admin context (should not be possible)', async () => {
    const mockStore = easiMockStore({
      groups: ['EASI_P_USER']
    });

    render(
      <MemoryRouter>
        <MockedProvider>
          <Provider store={mockStore}>
            <MessageProvider>
              <DocumentUploadForm type="admin" />
            </MessageProvider>
          </Provider>
        </MockedProvider>
      </MemoryRouter>
    );

    expect(
      screen.queryByText(
        'Would you like to send a notification email to GRB reviewers?'
      )
    ).not.toBeInTheDocument();
  });
});
