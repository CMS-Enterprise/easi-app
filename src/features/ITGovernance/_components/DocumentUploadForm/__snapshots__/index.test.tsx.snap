// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DocumentUploadForm page > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="tablet:grid-col-12 desktop:grid-col-8 margin-top-6 margin-bottom-8 padding-bottom-4"
  >
    <h1
      class="margin-bottom-1"
    >
      Upload a document
    </h1>
    <p
      class="margin-top-1 margin-bottom-1 font-body-md line-height-body-5 text-light"
    >
      Choose a document to upload such as a draft IGCE, contracting document, RAF or other document related to this project and Intake Request.
    </p>
    <p
      class="margin-bottom-3 margin-top-105 text-base"
    >
      Fields marked with an asterisk ( 
      <span
        class="text-red"
      >
        *
      </span>
       ) are required.
    </p>
    <a
      class="usa-link display-flex flex-align-center"
      href="/it-governance/63448b98-f92a-4227-bbf3-d07961ce242d/grb-review"
    >
      <svg
        aria-hidden="true"
        class="usa-icon margin-right-1"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      Don't upload and return to request details
    </a>
    <form
      class="usa-form maxw-full"
      data-testid="form"
    >
      <div
        class="usa-form-group margin-top-5"
        data-testid="formGroup"
      >
        <label
          class="usa-label"
          for="fileData"
        >
          Select your document
          <span
            class="text-red"
          >
             *
          </span>
        </label>
        <p
          class="usa-hint margin-bottom-0 margin-top-05"
        >
          Select a PDF, DOC, DOCX, XLS, or XLSX
        </p>
        <div
          class="usa-file-input"
          data-testid="file-input"
        >
          <div
            class="usa-file-input__target"
            data-testid="file-input-droptarget"
          >
            <div
              aria-hidden="true"
              class="usa-file-input__instructions"
              data-testid="file-input-instructions"
            >
              <span
                class="usa-file-input__drag-text"
              >
                Drag file here or 
              </span>
              <span
                class="usa-file-input__choose"
              >
                choose from folder
              </span>
            </div>
            <div
              class="usa-file-input__box"
              data-testid="file-input-box"
            />
            <input
              accept=".pdf,.doc,.docx,.xls,.xlsx"
              class="usa-file-input__input"
              data-testid="file-input-input"
              id="fileData"
              name="fileData"
              type="file"
            />
          </div>
        </div>
      </div>
      <div
        class="usa-form-group"
        data-testid="formGroup"
      >
        <fieldset
          class="usa-fieldset"
          data-testid="fieldset"
        >
          <legend
            class="usa-legend"
          >
            <span
              class="text-bold"
            >
              What type of document are you uploading? 
              <span
                class="text-red"
              >
                *
              </span>
            </span>
          </legend>
          <div
            class="usa-radio"
            data-testid="radio"
          >
            <input
              class="usa-radio__input"
              data-testid="documentType-SOO_SOW"
              id="documentType-SOO_SOW"
              name="documentType"
              type="radio"
              value="SOO_SOW"
            />
            <label
              class="usa-radio__label"
              for="documentType-SOO_SOW"
            >
              Statement of Objectives (SOO), Statement of Work (SOW), Performance Work Statement (PWS), or other contracting document
            </label>
          </div>
          <div
            class="usa-radio"
            data-testid="radio"
          >
            <input
              class="usa-radio__input"
              data-testid="documentType-DRAFT_IGCE"
              id="documentType-DRAFT_IGCE"
              name="documentType"
              type="radio"
              value="DRAFT_IGCE"
            />
            <label
              class="usa-radio__label"
              for="documentType-DRAFT_IGCE"
            >
              Draft Independent Government Cost Estimate (IGCE)
            </label>
          </div>
          <div
            class="usa-radio"
            data-testid="radio"
          >
            <input
              class="usa-radio__input"
              data-testid="documentType-ACQUISITION_PLAN_OR_STRATEGY"
              id="documentType-ACQUISITION_PLAN_OR_STRATEGY"
              name="documentType"
              type="radio"
              value="ACQUISITION_PLAN_OR_STRATEGY"
            />
            <label
              class="usa-radio__label"
              for="documentType-ACQUISITION_PLAN_OR_STRATEGY"
            >
              Acquisition Plan (AP) or Acquisition Strategy (AS)
            </label>
          </div>
          <div
            class="usa-radio"
            data-testid="radio"
          >
            <input
              class="usa-radio__input"
              data-testid="documentType-REQUEST_FOR_ADDITIONAL_FUNDING"
              id="documentType-REQUEST_FOR_ADDITIONAL_FUNDING"
              name="documentType"
              type="radio"
              value="REQUEST_FOR_ADDITIONAL_FUNDING"
            />
            <label
              class="usa-radio__label"
              for="documentType-REQUEST_FOR_ADDITIONAL_FUNDING"
            >
              Request for Additional Funding (RAF)
            </label>
          </div>
          <div
            class="usa-radio"
            data-testid="radio"
          >
            <input
              class="usa-radio__input"
              data-testid="documentType-SOFTWARE_BILL_OF_MATERIALS"
              id="documentType-SOFTWARE_BILL_OF_MATERIALS"
              name="documentType"
              type="radio"
              value="SOFTWARE_BILL_OF_MATERIALS"
            />
            <label
              class="usa-radio__label"
              for="documentType-SOFTWARE_BILL_OF_MATERIALS"
            >
              Software Bill of Materials (BOM)
            </label>
          </div>
          <div
            class="usa-radio"
            data-testid="radio"
          >
            <input
              class="usa-radio__input"
              data-testid="documentType-MEETING_MINUTES"
              id="documentType-MEETING_MINUTES"
              name="documentType"
              type="radio"
              value="MEETING_MINUTES"
            />
            <label
              class="usa-radio__label"
              for="documentType-MEETING_MINUTES"
            >
              Meeting Minutes
            </label>
          </div>
          <div
            class="usa-radio"
            data-testid="radio"
          >
            <input
              class="usa-radio__input"
              data-testid="documentType-OTHER"
              id="documentType-OTHER"
              name="documentType"
              type="radio"
              value="OTHER"
            />
            <label
              class="usa-radio__label"
              for="documentType-OTHER"
            >
              Other
            </label>
          </div>
        </fieldset>
      </div>
      <div
        class="usa-form-group"
        data-testid="formGroup"
      >
        <fieldset
          class="usa-fieldset"
          data-testid="fieldset"
        >
          <legend
            class="usa-legend"
          >
            <span
              class="text-bold"
            >
              What is the version of this document? 
            </span>
          </legend>
          <div
            class="usa-radio"
            data-testid="radio"
          >
            <input
              aria-describedby="versionHelpTextCURRENT"
              class="usa-radio__input"
              data-testid="version-CURRENT"
              id="version-CURRENT"
              name="version"
              type="radio"
              value="CURRENT"
            />
            <label
              class="usa-radio__label"
              for="version-CURRENT"
            >
              Current
            </label>
          </div>
          <p
            class="margin-left-4 margin-top-05 margin-bottom-0 font-body-2xs line-height-body-3"
            id="versionHelpTextCURRENT"
          >
            Choose this option if this is the most recent document version that the Governance Team should reference.
          </p>
          <div
            class="usa-radio"
            data-testid="radio"
          >
            <input
              aria-describedby="versionHelpTextHISTORICAL"
              class="usa-radio__input"
              data-testid="version-HISTORICAL"
              id="version-HISTORICAL"
              name="version"
              type="radio"
              value="HISTORICAL"
            />
            <label
              class="usa-radio__label"
              for="version-HISTORICAL"
            >
              Historical
            </label>
          </div>
          <p
            class="margin-left-4 margin-top-05 margin-bottom-0 font-body-2xs line-height-body-3"
            id="versionHelpTextHISTORICAL"
          >
            Choose this option if you are uploading a document from the past that should be used for reference purposes only.
          </p>
        </fieldset>
      </div>
      <div
        class="usa-form-group margin-top-3"
        data-testid="formGroup"
      >
        <fieldset
          class="usa-fieldset"
          data-testid="fieldset"
        >
          <legend
            class="usa-legend"
          >
            <span
              class="text-bold"
            >
              Would you like to send a notification email to GRB reviewers?
              <span
                class="text-red margin-x-05"
              >
                *
              </span>
            </span>
          </legend>
          <div
            class="easi-help-text margin-top-05"
          >
            If you choose “Yes”, EASi will automatically send a notification email to all GRB reviewers added to this request. Sending a notification while a GRB review is still in progress will help GRB reviewers stay up to date with all relevant project materials.
          </div>
          <div
            class="usa-radio"
            data-testid="radio"
          >
            <input
              class="usa-radio__input"
              id="sendNotification-yes"
              name="sendNotification"
              type="radio"
            />
            <label
              class="usa-radio__label"
              for="sendNotification-yes"
            >
              Yes
            </label>
          </div>
          <div
            class="usa-radio"
            data-testid="radio"
          >
            <input
              class="usa-radio__input"
              id="sendNotification-no"
              name="sendNotification"
              type="radio"
            />
            <label
              class="usa-radio__label"
              for="sendNotification-no"
            >
              No
            </label>
          </div>
        </fieldset>
      </div>
      <div
        class="usa-alert usa-alert--info usa-alert--slim margin-top-5"
        data-testid="alert"
      >
        <div
          class="usa-alert__body"
        >
          <p
            class="usa-alert__text"
          >
            To keep CMS safe, documents are scanned for viruses after uploading. If something goes wrong, we'll let you know.
          </p>
        </div>
      </div>
      <div
        class="margin-top-4"
      >
        <ul
          class="usa-button-group"
        >
          <li
            class="usa-button-group__item"
          >
            <div
              class="display-flex flex-align-center"
            >
              <button
                class="usa-button margin-top-0"
                data-testid="pager-next-button"
                disabled=""
                type="submit"
              >
                Upload document
              </button>
            </div>
          </li>
        </ul>
        <button
          class="usa-button usa-button--unstyled margin-top-2 display-flex flex-align-center"
          data-testid="button"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="usa-icon margin-right-05"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
          Don't upload and return to request details
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
